src/signer_node/message_processor/process_blocksig.rs

```diff
@@ -158,3 +158,380 @@ where
+            blockhash,
+            gamma_i,
+            e,
+            &priv_shared_key,
+            &shared_secrets,
+            &prev_state,
+            &conman,
+            &params,
+        );
+
+        assert_eq!(next, prev_state);
+    }
+
+    #[test]
+    fn test_process_blocksig_receiving_invaid_signature() {
+        // when node
+        //  - receives a valid block,
+        //  - has the number of signatures to generate a aggregated signature,
+        //  - but received gamma_i and e is invalid.
+        // node should return prev_state.
+
+        let contents = load_test_vector("./tests/resources/test_vectors.json").unwrap();
+
+        let conman = TestConnectionManager::new();
+        let rpc = MockRpc::new();
+        let (signers, blockhash, gamma_i, e, priv_shared_key, shared_secrets, prev_state, params) =
+            load_test_case(
+                &contents,
+                "process_blocksig_receiving_invaid_signature",
+                rpc,
+            );
+
+        let next = process_blocksig(
+            &signers[0],
+            blockhash,
+            gamma_i,
+            e,
+            &priv_shared_key,
+            &shared_secrets,
+            &prev_state,
+            &conman,
+            &params,
+        );
+
+        assert_eq!(next, prev_state);
+    }
+
+    #[test]
+    fn test_process_blocksig_with_invaid_signature() {
+        // when node
+        //  - receives a valid block,
+        //  - has the number of signatures to generate a aggregated signature,
+        //  - and received gamma_i and e is valid.
+        //  - but node already received invalid signature from other node
+        // node should return prev_state.
+
+        let contents = load_test_vector("./tests/resources/test_vectors.json").unwrap();
+
+        let conman = TestConnectionManager::new();
+        let rpc = MockRpc::new();
+        let (signers, blockhash, gamma_i, e, priv_shared_key, shared_secrets, prev_state, params) =
+            load_test_case(&contents, "process_blocksig_with_invaid_signature", rpc);
+
+        let next = process_blocksig(
+            &signers[0],
+            blockhash,
+            gamma_i,
+            e,
+            &priv_shared_key,
+            &shared_secrets,
+            &prev_state,
+            &conman,
+            &params,
+        );
+
+        assert_eq!(next, prev_state);
+    }
+
+    #[test]
+    fn test_process_blocksig_successfully() {
+        // when node
+        //  - receives a valid block,
+        //  - has the number of signatures to generate a aggregated signature,
+        //  - has block key,
+        // then node should
+        //  - call rpc submitblock
+        //  - send message `Completedblock`
+        //  - return Master
+
+        let contents = load_test_vector("./tests/resources/test_vectors.json").unwrap();
+
+        let conman = TestConnectionManager::new();
+        let mut rpc = MockRpc::new();
+        rpc.should_call_submitblock(Ok(()));
```

Comment:
You should call `rpc.assert();` after finished using rpc mock for checking there are any remaining stub.
See document in `src/tests/helper/rpc.rs`

Reply:
it is called at line:410

Reply:
Oh sorry, I missed it.